### PR TITLE
fix ingress delete panic

### PR DESCRIPTION
**Pull request by champly**
Signed-off-by: champly <champly@outlook.com>

### Type of change:

- [x] Bugfix
- [ ] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:
#855 

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by champly** [comment]
@tao12345666333 PTAL

**Comment by champly** [comment]
> Hi, thanks for the PR. I believe we should also change the error log for `errLocal` as it's no longer a split meta key error.

Already record `errLocal` log, PTAL.

**Comment by lingsamuel** [comment]
Thank you for the update.
It would be good to change the logs for L222 and L246 as well.

**Comment by champly** [comment]
> Thank you for the update. It would be good to change the logs for L222 and L246 as well. These two lines record the actual error.

Since there is `key` information inside, only `errLocal` is printed outside. PTAL

File: pkg/ingress/ingress.go [hunk 1]
```diff
@@ -192,18 +192,20 @@ func (c *ingressController) handleSyncErr(obj interface{}, err error) {
 	event := ev.Object.(kube.IngressEvent)
 	namespace, name, errLocal := cache.SplitMetaNamespaceKey(event.Key)
 	if errLocal != nil {
-		log.Errorf("invalid resource key: %s", event.Key)
+		log.Errorw("invalid resource key",
```
Comment (tokers) [inline pkg/ingress/ingress.go:195]: I think still reserve the `key` in the log will be better.

Comment (champly) [reply pkg/ingress/ingress.go:195]: `cache.SplitMetaNamespaceKey` return `errLocal` include `key`
``` go
return "", "", fmt.Errorf("unexpected key format: %q", key)
```

**Review by gxthrj — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Review by gxthrj — APPROVED** [review]

**Comment by tao12345666333** [comment]
Thanks

**Comment by lijing-21** [comment]
Hi @champly , thank you for your contribution!

Here is the Contributor T-shirt form[1], if you're interested, kindly take a look :)

[1] https://github.com/apache/apisix/blob/master/CONTRIBUTING.md#contributor-t-shirt

**Pull request merged**